flask_cors/core.py

```diff
@@ -226,6 +226,14 @@ def set_cors_headers(resp, options):
         LOG.debug('CORS have been already evaluated, skipping')
         return resp
 
+    # If resp.headers is not a MultiDict, set resp.headers to a new
+    # MultiDict, copying the underlying dict.
+    if type(resp.headers) is MultiDict:
```

Comment:
Awesome start!

Three quick pieces of feedback:
1. It is more Pythonic to check isinstance, e.g. `isinstance(resp.headers, MultiDict)`, since anything that extends MultiDict will also (by extension) have the methods we need.
2. I think this check is inverted. If headers is not a MultiDict, then we want to set it to be a MultiDict.
3. Instead of manually iterating the `headers` object, we can let MultiDict do the work for us by passing it to the constructor.

I think combined, it would simplify a little bit to something like:

```Python

# OauthLib  sets resp.headers to a non MultiDict. Weird, but let's work around it. 
if not isinstance(resp.headers, MultiDict):
    resp.headers = MultiDict(resp.headers)
```